### ISSUE TITLE
nspawn options ignored if on different disk 

### ISSUE THREAD

**Issue by damianatorrpm** [systemd/systemd#16966]
Hi,
I'm on latest Silverblue 33.
Previously I had success running a nspawn container that allows specific /dev interaction (needed by [OBS](https://openbuildservice.org/)).
`sudo systemd-nspawn -D $TARGETDIR -b --resolv-conf=bind-host --capability=all --property="DeviceAllow=/dev/null rwm" --property="DeviceAllow=/dev/zero rwm" --property="DeviceAllow=/dev/random rwm" --property="DeviceAllow=/dev/urandom rwm" --property="DeviceAllow=/dev/loop-control rwm" --property="DeviceAllow=/dev/loop-* rwm" --property="DeviceAllow=/dev/port rwm" --property="DeviceAllow=block-blkext rwm" --property="DeviceAllow=char-pts rwm"  --property="DeviceAllow=block-loop rwm" --property="DeviceAllow=block-* rwm" --bind="$RUNDIR/obs/:/run/obs" --bind="$RUNDIR/passenger/:/run/passenger/" --bind="$RUNDIR/mysql/:/run/mysql/" --rlimit=RLIMIT_NOFILE=8192:65536 &`
That was all working fine until I reinstalled the system and tried to symlink from the original partition to a folder on the new partition (/home/tux/OBS_CONTAINER, the $TARGETDIR, on the new installation is a symlink to /mountPointOfOldSystem/home/tux/OBS_CONTAINER)

I found that now various device accesses failed inside the nspawn (like /dev/null).
If instead of symlinking the directory is copied via rsync -aHAXv oldDir newDir it works.

It seems that the special DeviceAllow properties are ignored if the $TARGETDIR is a symlink and the symlink target is living on a different SSD.

**Comment by poettering** [comment]
do the specified paths have the same basenames? i.e. the rightmost component of the paths is the same?

Note that /dev/null, /dev/zero and so on are made available to containers anyway, they are redundant to specify.

**Comment by poettering** [comment]
what systemd version is this?

**Comment by damianatorrpm** [comment]
systemd 246 (v246.4-1.fc34)
I'm not sure I completely understand the basename question.
On the original installation I had
/home/tux/OBSCONTAINER (normal directory)

Than I installed a new system on a different ssd
There I symlinked 
/home/tux/OBSCONTAINER 
to the different ssd with the link target being /home/tux/OBSCONTAINER on the other drive.

Starting from the new installation the /home/tux/OBSCONTAINER it says /dev/null permission denied,
when building packages in OBS.

This doesn't happen if it is not symlink.
